tests/bluetooth/host_long_adv_recv/src/main.c

```diff
@@ -427,49 +423,58 @@ ZTEST(long_adv_rx_tests, test_host_long_adv_recv)
 		     report_b_2.length);
 
 	/* Check that non-interleaved fragmented adv reports work */
-	ztest_returns_value(get_expected_data, report_a_combined.data);
-	ztest_returns_value(get_expected_length, report_a_combined.length); /* Expect a */
-	ztest_returns_value(get_expected_data, report_b_combined.data);
-	ztest_returns_value(get_expected_length, report_b_combined.length); /* Then b */
+	expected_reports[0] = report_a_combined;
+	expected_reports[1] = report_b_combined;
+	SET_RETURN_SEQ(get_expected_report, expected_reports, 2);
 	send_adv_report(&report_a_1);
 	send_adv_report(&report_a_2);
 	send_adv_report(&report_b_1);
 	send_adv_report(&report_b_2);
+	zassert_equal(2, get_expected_report_fake.call_count, NULL);
+	RESET_FAKE(get_expected_report);
+	FFF_RESET_HISTORY();
```

Comment:
I think it is safer to have the resets before each test, not after. (There are several tests in this function in this manner of speaking.) The FFF examples put it in a "setup" function. See this [FFF example].

[FFF example]: https://github.com/meekrosoft/fff/blob/master/examples/embedded_ui/UI_test_ansic.c#L31

```suggestion
	/* Setup: Reset all FFF state. */
	RESET_FAKE(get_expected_report);
	FFF_RESET_HISTORY();

	/* Test: Check that non-interleaved fragmented adv reports work */
	ztest_returns_value(get_expected_data, report_a_combined.data);
	ztest_returns_value(get_expected_length, report_a_combined.length); /* Expect a */
	ztest_returns_value(get_expected_data, report_b_combined.data);
	ztest_returns_value(get_expected_length, report_b_combined.length); /* Then b */
	expected_reports[0] = report_a_combined;
	expected_reports[1] = report_b_combined;
	SET_RETURN_SEQ(get_expected_report, expected_reports, 2);
	send_adv_report(&report_a_1);
	send_adv_report(&report_a_2);
	send_adv_report(&report_b_1);
	send_adv_report(&report_b_2);
	zassert_equal(2, get_expected_report_fake.call_count, NULL);
```

Reply:
I'll be happy to do so if you feel strongly about it, the reason for resetting in the middle of the function was for readability and trying to keep the tests with as small a diff as possible. The test you commented on is really testing multiple things and should have been multiple tests, but wasn't. It's possible to remove the resets and set a much longer return sequence as well as deeper asserts (that check the full call history at the end of the test) but I would assume that the owners of the test would like to eventually split this up, so I kept it as similar to the original test as possible.